### PR TITLE
General polishing

### PR DESCRIPTION
**Pull request by sabbyanandan**
## Description

As I was reviewing the public-facing documentation, I had a few suggestions as incremental polishing, which I captured in this PR.

At a high-level,
- reorganize the getting-started section of the left-nav
- correct the references to Stacks Node API vs. Stacks Blockchain API for consistency
- add a bit more color to the install scripts that trumped me 
- copy+paste install steps from README for Clarinet (couldn't find a way to include README fragments directly)
- fix several broken links
- polish language

Outside of this, a few sample apps need a bit of TLC—screenshots are outdated (hiro wallet had evolved), and some don't work (eg: heystack). I will capture separate issues for them.

## Checklist

- [x] For fixes/refactors: all existing references were identified and replaced
- [x] People were tagged for review

### CODE REVIEW

File: docs/build-apps/overview.md [hunk 1]
```diff
@@ -15,6 +15,6 @@ Three main integrations are available:
 - **Transaction signing**: Prompt users to sign and broadcast transactions to the Stacks blockchain
 - **Data storage**: Save and retrieve data for users with [Gaia](https://docs.stacks.co/build-apps/references/gaia)
 
-All three of these integrations can be used together to create powerful new user experiences that rival or exceed those of traditional apps—all while protecting your users' digital rights.
+All three of these integrations can be used together to create powerful new user experiences that rival or exceed traditional apps—all while protecting your users' digital rights.
```
Comment (diwakergupta) [inline docs/build-apps/overview.md:18]: [nit] why delete "those of"? Without that, it's unclear what is the subject of "exceed" -- experiences exceeding apps sounds odd, experiences exceeding those of traditional apps made sense.

File: docs/get-started/running-api-node.md [hunk 2]
```diff
@@ -3,8 +3,8 @@ id: running-api-node
 title: Running a Stacks API node
 ---
 
-This procedure demonstrates how to run a local API node using Docker images. There are several components that must be
-configured and run in a specific order for the local API node to work.
+This procedure demonstrates how to run a local Stacks Blockchain API node using Docker images. There are several components that must be
```
Comment (diwakergupta) [inline docs/get-started/running-api-node.md:6]: I think how we refer to the API maybe warrants some discussion before we start making changes. Like the title says "Stacks API" but the text says "Stacks Blockchain API". There's already confusion in the endpoints provided by the API vs. the endpoints provided by the Stacks blockchain nodes (the /v2 endpoints)

File: docs/get-started/running-api-node.md [hunk 3]
```diff
@@ -87,7 +86,7 @@ docker run -d --rm \
   --net=stacks-blockchain \
   -e POSTGRES_PASSWORD=postgres \
   -v $(pwd)/persistent-data/postgres:/var/lib/postgresql/data \
-  -p 5432:5432
+  -p 5432:5432 \
```
Comment (diwakergupta) [inline docs/get-started/running-api-node.md:89]: I believe this was fixed in a separate PR already, might have to resolve conflicts

File: docs/get-started/stacks-blockchain-api.md [hunk 4]
```diff
@@ -7,11 +7,11 @@ The Stacks 2.0 blockchain API allows you to query the Stacks 2.0 blockchain and
 
 :::caution
 
-This API is hosted by Hiro. Using it requires you to trust the hosted server, but provides a faster development experience. Consider [running your own API instance](#running-an-api-server) to create a fully trustless architecture for your app.
+Hiro hosts the Stacks 2.0 blockchain API. Using it requires you to trust the hosted server, providing a faster development experience. Consider [running your own API instance](#running-an-api-server) to create a fully trustless architecture for your app.
```
Comment (diwakergupta) [inline docs/get-started/stacks-blockchain-api.md:10]: Maybe say "Hiro hosts an instance of ..." to clarify that there might be instances hosted by others?

File: docs/get-started/stacks-blockchain-api.md [hunk 5]
```diff
@@ -24,7 +24,7 @@ Review the [Stacks API reference](/api) for more details
 
 :::
 
-The API is comprised of two parts: the Stacks Blockchain API and the Stacks Node RPC API. The Node RPC API is exposed by every running node. Stacks Blockchain API, however, introduces additional capabilities (for example, retrieving all transactions). It also proxies calls directly to Stacks Node RPC API.
+The Stacks 2.0 Blockchain API is comprised of two parts: the "Stacks Blockchain API" and the "Stacks Node RPC API". The Stacks Node RPC API is exposed by every running node participating in the Stacks Blokchain. However, the Stacks Blockchain API introduces additional capabilities (for example, retrieving all transactions). It also proxies calls directly to Stacks Node RPC API.
```
Comment (diwakergupta) [inline docs/get-started/stacks-blockchain-api.md:27]: Even in this PR, we're using "Stacks 2.0 Blockchain API" in some places and "Stacks Blockchain API" in other places. Also need to standardize on capitalization -- blockchain needn't be capitalized, for instance.

File: docs/build-apps/overview.md (hunk 1 above)
Comment (sabbyanandan) [reply docs/build-apps/overview.md:18]: Nothing technical other than attempting to tightening up the language. I will revert this change.

**Comment by sabbyanandan** [comment]
@diwakergupta: Please let me know if you see any other odd bits.

**Pull request closed without merging**